### PR TITLE
add: user dropdown

### PR DESCRIPTION
**Pull request by kodaline**
# Description

Added a user dropdown to manage logout, theme change

Current logged username is displayed as additional info

refers to issue #93

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings

### CODE REVIEW

File: src/stores/useMainStore.ts [hunk 1]
```diff
@@ -83,12 +83,24 @@ export const useMainStore = defineStore('main', () => {
 		},
 	})
 
+	const logoutCurrentUser = () => {
+		const cookies = document.cookie.split(";");
```
Comment (zAlweNy26) [inline src/stores/useMainStore.ts:87]: You can use the `useCookies` composable for managing with cookies :)

File: src/stores/useMainStore.ts [hunk 2]
```diff
@@ -83,12 +83,24 @@ export const useMainStore = defineStore('main', () => {
...
+
+		cookies.forEach(cookie => {
+			const name = cookie.split("=")[0].trim();
+			document.cookie = `${name}=;path=/;domain=${window.location.hostname};`;
+		})
+		cookie.value = ''
+		window.location.href = window.location.origin + '/auth/login' // this is horrible coding, TODO also use parameter for /auth/login
```
Comment (zAlweNy26) [inline src/stores/useMainStore.ts:94]: You should use the `useRouter` composable for this :)

File: src/components/ThemeButton.vue [hunk 3]
```diff
@@ -15,9 +14,9 @@ watchEffect(() => {
 </script>
 
 <template>
-	<button class="btn btn-circle btn-ghost swap" @click="toggleDark()">
```
Comment (zAlweNy26) [inline src/components/ThemeButton.vue:18]: As this component is not used anywhere else, I think we can integrate it inside the `UserDropdown` component you created :)

File: src/components/Logout.vue [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<script setup lang="ts">
+import { useMainStore } from '@/stores/useMainStore';
+
+const store = useMainStore()
+const { logoutCurrentUser } = store
+
+</script>
+
+<template>
+	<button class="opacity-50" @click="logoutCurrentUser()">
```
Comment (zAlweNy26) [inline src/components/Logout.vue:10]: It's better to integrate it in the component you created for the dropdown, creating a component just for this button seems too much :/

**Review by zAlweNy26 — CHANGES_REQUESTED** [review]

File: src/stores/useMainStore.ts (hunk 2 above)
Comment (kodaline) [reply src/stores/useMainStore.ts]: I think for this one we need to find a better solution, trying router.push('/auth/login') does not work, maybe I tried wrong

**Comment by zAlweNy26** [comment]
can you please rebase from develop?

**Pull request merged**